packages/Search/src/details/DTK_DetailsKokkosExt.hpp

```diff
@@ -32,6 +42,97 @@ struct is_accessible_from_host<
     : std::true_type
 {
 };
+
+/** Count the number of consecutive leading zero bits in 32 bit integer
+ * @param x
+ */
+KOKKOS_INLINE_FUNCTION
+int clz( uint32_t x )
+{
+#if defined( __CUDA_ARCH__ )
+    // Note that the __clz() CUDA intrinsic function takes a signed integer
+    // as input parameter.  This is fine but would need to be adjusted if
+    // we were to change expandBits() and morton3D() to subdivide [0, 1]^3
+    // into more 1024^3 bins.
+    return __clz( x );
+#elif defined( KOKKOS_COMPILER_GNU ) || ( KOKKOS_COMPILER_CLANG >= 500 )
+    // According to https://en.wikipedia.org/wiki/Find_first_set
+    // Clang 5.X supports the builtin function with the same syntax as GCC
+    return ( x == 0 ) ? 32 : __builtin_clz( x );
+#else
+    if ( x == 0 )
+        return 32;
+    // The following is taken from:
+    // http://stackoverflow.com/questions/23856596/counting-leading-zeros-in-a-32-bit-unsigned-integer-with-best-algorithm-in-c-pro
+    const char debruijn32[32] = {0,  31, 9,  30, 3,  8,  13, 29, 2,  5, 7,
+                                 21, 12, 24, 28, 19, 1,  10, 4,  14, 6, 22,
+                                 25, 20, 11, 15, 23, 26, 16, 27, 17, 18};
+    x |= x >> 1;
+    x |= x >> 2;
+    x |= x >> 4;
+    x |= x >> 8;
+    x |= x >> 16;
+    x++;
+    return debruijn32[x * 0x076be629 >> 27];
+#endif
+}
+
+//! Compute the maximum of two values.
+template <typename T, typename = std::enable_if_t<std::is_arithmetic<T>::value>>
```

Comment:
Now I realize this is dumb to restrict to arithmetic types :/

Reply:
The tests passed so I am going to merge as is and fix in an upcoming PR where I remove our use of `Kokkos::ArithTraits` altogether.